### ISSUE TITLE
embed error

### ISSUE THREAD

**Issue by mikemliu** [lovell/sharp#204]
Getting an error embedding png with transparency on the default black background (inside larger background canvas)

start with a 50x50 png with alpha channel:
```
sharp(in)
.resize(50, 256)
.embed()
.pipe(out)
```
i get the following:
```
Error: linear: vector must have 1 or 4 elements
```

The following DO work:
- embedding a png with no transparency
- embedding on a background that matches the source aspect ratio

**Comment by mikemliu** [comment]
Flatten seems to resolve the issue:
```
sharp(in)
.resize(50, 256)
.flatten()
.embed()
.pipe(out)
```

**Comment by lovell** [comment]
Looks like a bug, thanks for reporting this. I'll try to add a test case to confirm this later today.

Were you hoping to preserve transparency in the output image?

**Comment by lovell** [comment]
I've reproduced this @mikemliu. It occurs when embedding an image with an alpha channel onto any background colour that does not have a transparency, which includes the default.

I've added a new test case to expose the failure and the fix in commit 33782d3.

Are you able to retest via `npm install lovell/sharp`?

**Comment by mikemliu** [comment]
I realized the same when I actually put in my desired transparent background. Thanks!

**Comment by lovell** [comment]
@mikemliu Were you able to confirm this fix solved the problem? If so I'll publish a point release with it to npm.

**Comment by mikemliu** [comment]
confirmed. thanks!

**Comment by lovell** [comment]
This is fixed in v0.10.1, now available via npm.
